### PR TITLE
test: add unit tests for CardUtils

### PR DESCRIPTION
**Pull request by Ayush-Patel-56**
## Purpose / Description
Added unit tests for CardUtils. This class had no test coverage before.

## Fixes
* Fixes part of #13283

## Approach
Created CardUtilsTest.kt with 7 test cases covering both public methods:

**`getNotes()`:**
- Empty collection handling
- Single card scenario
- Multiple cards from same note (verifies unique filtering)
- Multiple cards from different notes

**`getDeckIdForCard()`:**
- Regular deck (oDid = 0)
- Filtered/cram deck (oDid handling)
- Explicit zero oDid edge case

## How Has This Been Tested?
Ran unit tests locally:
./gradlew :AnkiDroid:testPlayDebugUnitTest --tests "com.ichi2.anki.CardUtilsTest"
All 7 tests passed with 0 failures.

## Learning (optional, can help others)
- Studied existing test patterns from ImportUtilsTest and NoteServiceTest to match project conventions
- Used RobolectricTest base class which provides helper methods like `addBasicNote()` and `addDynamicDeck()`
- Learned that filtered decks use the `oDid` field to store the original deck ID when cards are moved to cram decks

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import org.hamcrest.Matchers.hasSize
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class CardUtilsTest : RobolectricTest() {
+    @Test
+    fun getNotes_emptyCollection_returnsEmptySet() {
+        val cards = emptyList<com.ichi2.anki.libanki.Card>()
+        val notes = CardUtils.getNotes(col, cards)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]: ⚠️ `getNotes` is unused, and should be deleted

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        assertThat("Should return the original deck ID (oDid)", deckId, equalTo(card.oDid))
+    }
+
+    @Test
+    fun getDeckIdForCard_zeroODid_returnsDid() {
+        val note = addBasicNote("Front", "Back")
+        val card = note.firstCard()
+
+        // Explicitly verify oDid is 0
+        card.oDid = 0L
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]: Assert/Assume on the property, don't set it.

Comment (Ayush-Patel-56) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]: Thanks for the review! Fixed - now using `assertThat()` to verify oDid is 0 instead of setting it.

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt (hunk 1 above)
Comment (Ayush-Patel-56) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]: Done - removed the trivial empty collection test.

Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]: The method in `CardUtils` should be deleted, not just the test.

Comment (Ayush-Patel-56) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]: Done, removed getNotes() from CardUtils.kt as well

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        assertThat("Should return the original deck ID (oDid)", deckId, equalTo(card.oDid))
+    }
+
+    @Test
+    fun getDeckIdForCard_zeroODid_returnsDid() {
+        val note = addBasicNote("Front", "Back")
+        val card = note.firstCard()
+
+        // Verify oDid is 0 for regular decks
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]:
```suggestion
```

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    fun getDeckIdForCard_cramDeck_returnsODid() {
+        val note = addBasicNote("Front", "Back")
+        val card = note.firstCard()
+
+        // Move card to a filtered deck
+        val filteredDid = addDynamicDeck("Filtered")
+        col.sched.rebuildFilteredDeck(filteredDid)
+        card.load()
+
+        // In a filtered deck, oDid should be set
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]:
```suggestion
```

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+        assertThat("Should return the card's did", deckId, equalTo(card.did))
+    }
+
+    @Test
+    fun getDeckIdForCard_cramDeck_returnsODid() {
+        val note = addBasicNote("Front", "Back")
+        val card = note.firstCard()
+
+        // Move card to a filtered deck
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]:
```suggestion
```

File: AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class CardUtilsTest : RobolectricTest() {
+    @Test
+    fun getDeckIdForCard_regularDeck_returnsDid() {
+        val note = addBasicNote("Front", "Back")
+        val card = note.firstCard()
+
+        // Regular deck has oDid = 0
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardUtilsTest.kt]:
```suggestion
```

**Comment by Ayush-Patel-56** [comment]
Fair point! I tend to over-comment everything as I'm still learning the codebase 😅, but I see how it's redundant here. I'll keep it in mind to avoid this in the future. Fixed the suggestions, thanks!
